code/Model/SiteTree.php

```diff
@@ -2730,6 +2730,17 @@ public function getControllerName()
         return $controller;
     }
 
+    /**
+     * Find appropriate templates for SSViewer to use to render this page
+     *
+     * @param string $suffix
+     * @return array
+     */
+    public function getViewerTemplates($suffix = '')
+    {
+        return SSViewer::get_templates_by_class(get_class($this), $suffix, self::class);
```

Comment:
As per my comment, maybe a ViewableData method that stops at self::class, and DataObject method that stops at `$this->baseClass()`